services/motion/builtin/builtin.go

```diff
@@ -327,8 +329,19 @@ func (ms *builtIn) MoveOnGlobe(
 			if resp.err != nil {
 				return false, resp.err
 			}
+			ms.logger.Info("obstacle detection triggering a replan")
+		}
+		if resp.replan {
```

Comment:
I need to think about this more but I don't think this belongs here.  The move request was not initially intended to be stateful and I wanted to capture everything in the select statement. Wondering what @nicksanford thinks

Reply:
The move request is not being used statefully here; it is being wholesale replaced with a new one.

Reply:
With the exception of `seedPlan` being set.

Reply:
Ok did some more thinking on this I like this a lot actually!  I think we can get rid of the conditional since if we're getting to this point we will definitionally be replanning. 

I see now that its not being done statefully but since that is not the case we are doing a lot of redundant work in creating these move requests since they will be exactly the same.  There is a good bit of work being done in this function so I feel like it could be a worthwhile optimization to not dupliate this work